### ISSUE TITLE
cannot access my nodejs application even adding a url

### ISSUE THREAD

**Issue by dremond71** [openshift/odo#2583]
/kind bug

I decided to try them 60 day evaluation version of RedHat OpenShift Online.

I figured a Node.js / Express.js  REST API application would suffice. I built it and tested it locally in and outside of Docker.

First I pulled in my app from github via the web ui, and everything got built and running.
Awesome.
BUT, for the life of me ...could not figure out HOW to access my running nodejs application
from postman. Nothing worked. 
I tried the IP address of the kubernetes cluster but it didn't seem like an external ip address. It started with 172.xxxxx .

I found documentation about externalIP something and I needed to access some master config file, but couldn't figure out what this was in the web ui.

Then I downloaded oc and odo .

I created a template nodejs app from the list of templates listed by odo.
The pod had issues with the image.

I deleted everything and started from scratch.

I then proceeded to follow the instructions from 
```
odo --help
```

e.g. 

```
git clone https://github.com/myRepositoryUrl
  odo create nodejs
  odo push
  
 
  odo url create
 odo push -f
```
and still cannot figure out what my external ip address is for the nodejs application.

```
odo url list 

```

returns 

```
> odo url list
Found the following URLs for component nodejs-rhosapp3-fogk in application app:
NAME      STATE      URL                                                                       PORT
myurl     Pushed     http://myurl-app-rhosapp6.apps.us-east-1.starter.openshift-online.com     8080
```

What is this url?  It just hangs...

When I list my service(s):

```
oc get svc
```

it returns 

```

NAME                       TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
nodejs-rhosapp3-fogk-app   ClusterIP   172.30.59.239   <none>        8080/TCP   17m
```

The external ip is still NONE

Any help is appreciated.

I've spent way too many hours trying to figure this out.

It should be easy to deploy a nodejs app to this platform, AND access it.

Any help is appreciated.

:)

## What versions of software are you using?

**Operating System:**

**Output of `odo version`:**

**Comment by kadel** [comment]
OpenShift Online is not allowing `NodePorts` or `LoadBalancer` Services, so you can't expose you application just using `Service` (there will be no externalIp when you run`oc get svc` ).

The url that you created using `odo url create` should work. It creates OpenShift Route resource, you should be able to see it when you run `odo get route`.
If it just hangs than something might be up with the nodejs application.

What do you get if you run `odo log`?

**Comment by dremond71** [comment]
This is my repo on github:

 https://github.com/dremond71/rhosapp3

**Comment by dremond71** [comment]
- I have a directory `db` with a json file (as a child of the main project directory)
- I have a Dockerfile

I have tested my app on my laptop in and outside a docker container and it works.

Perhaps I am not supposed to have a Dockerfile, or it has to specify specific things?

**Comment by dremond71** [comment]
odo log

```
2020/02/11 01:21:08 [go-init] Pre-start command launched : /opt/odo/bin/s2i-setup
+ set -eo pipefail
+ PV_MNT_PT=/opt/app-root
+ ODO_UTILS_DIR=/opt/odo
+ '[' '!' -f /opt/app-root/conf/supervisor.conf ']'
+ cp -rp /opt/odo/conf /opt/app-root
+ mkdir -p /tmp
+ touch /tmp/.dummy
+ mkdir -p /tmp/src/
+ touch /tmp/src/.dummy
+ [[ /tmp != *\/\s\r\c* ]]
+ mkdir -p /tmp/src
+ touch /tmp/src/.dummy
+ mkdir -p /opt/app-root/src
2020/02/11 01:21:08 [go-init] Pre-start command exited
2020/02/11 01:21:08 [go-init] Main command launched : /opt/odo/bin/supervisord -c /opt/app-root/conf/supervisor.conf     
time="2020-02-11T01:21:08Z" level=info msg="create process:run"
time="2020-02-11T01:21:50Z" level=debug msg="no auth required"
time="2020-02-11T01:21:50Z" level=info msg="stop process" program=run
time="2020-02-11T01:21:50Z" level=debug msg="succeed to find process:run"
time="2020-02-11T01:21:50Z" level=info msg="stop the program" program=run
time="2020-02-11T01:21:50Z" level=info msg="force to kill the program" program=run
time="2020-02-11T01:21:50Z" level=debug msg="no auth required"
time="2020-02-11T01:21:50Z" level=debug msg="succeed to find process:run"
time="2020-02-11T01:21:50Z" level=info msg="try to start program" program=run
time="2020-02-11T01:21:50Z" level=info msg="success to start program" program=run
+ set -eo pipefail
+ export ODO_UTILS_DIR=/opt/odo
+ ODO_UTILS_DIR=/opt/odo
+ export ODO_IMAGE_MAPPINGS_FILE=/opt/odo/language-scripts/image-mappings.json
+ ODO_IMAGE_MAPPINGS_FILE=/opt/odo/language-scripts/image-mappings.json
++ /opt/odo/bin/getlanguage
+ IMAGE_LANG=nodejs
+ '[' -f /tmp/.s2i/bin/run ']'
+ '[' '!' -z nodejs ']'
+ exec /opt/odo/language-scripts/nodejs/dev-run
+ DEV_MODE=true
+ '[' -z '' ']'
+ '[' true == true ']'
+ export NODE_ENV=development
+ NODE_ENV=development
+ '[' development == production ']'
+ export DEV_MODE=true
+ DEV_MODE=true
++ id -u
+ USER_ID=1008510000
++ id -g
+ GROUP_ID=0
+ '[' x1008510000 '!=' x0 -a x1008510000 '!=' x1001 ']'
+ NSS_WRAPPER_PASSWD=/opt/app-root/etc/passwd
+ NSS_WRAPPER_GROUP=/etc/group
+ sed -e 's/^default:/builder:/'
+ cat /etc/passwd
+ echo 'default:x:1008510000:0:Default Application User:/opt/app-root/src:/sbin/nologin'
+ export NSS_WRAPPER_PASSWD
+ export NSS_WRAPPER_GROUP
+ LD_PRELOAD=libnss_wrapper.so
+ export LD_PRELOAD
+ '[' development == production ']'
+ export DEV_MODE=true
+ DEV_MODE=true
+ set -x
+ export GIT_COMMITTER_NAME=unknown
+ GIT_COMMITTER_NAME=unknown
+ export and GIT_COMMITTER_EMAIL=unknown@localhost.com
+ GIT_COMMITTER_EMAIL=unknown@localhost.com
+ git --version
git version 1.8.3.1
+ '[' '' == --debug ']'
+ run_node
++ node --version
Using Node.js version: v10.16.3
+ echo -e 'Using Node.js version: v10.16.3'
Environment:
        DEV_MODE=true
        NODE_ENV=development
        DEBUG_PORT=5858
+ echo -e 'Environment: \n\tDEV_MODE=true\n\tNODE_ENV=development\n\tDEBUG_PORT=5858'
++ id
+ echo -e 'Running as user uid=1008510000(1008510000) gid=0(root) groups=0(root),1008510000'
Running as user uid=1008510000(1008510000) gid=0(root) groups=0(root),1008510000
+ '[' true == true ']'
+ echo 'Launching via nodemon...'
Launching via nodemon...
+ exec npx nodemon --inspect=5858
time="2020-02-11T01:21:51Z" level=debug msg="wait program exit" program=run
[nodemon] 2.0.2
[nodemon] to restart at any time, enter `rs`
[nodemon] watching dir(s): *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node --inspect=5858 index.js`
Debugger listening on ws://127.0.0.1:5858/65f654b3-a6a4-4d62-bf6a-7cbe9734d37f
For help, see: https://nodejs.org/en/docs/inspector
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module '/opt/app-root/src/index.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
[nodemon] app crashed - waiting for file changes before starting...
[nodemon] restarting due to changes...
[nodemon] starting `node --inspect=5858 index.js`
[nodemon] restarting due to changes...
[nodemon] restarting due to changes...
[nodemon] starting `node --inspect=5858 index.js`
Debugger listening on ws://127.0.0.1:5858/0b62b062-2754-4922-b12b-dfe3f7b79814
For help, see: https://nodejs.org/en/docs/inspector
[nodemon] restarting due to changes...
[nodemon] restarting due to changes...
[nodemon] starting `node --inspect=5858 index.js`
[nodemon] restarting due to changes...
[nodemon] starting `node --inspect=5858 index.js`
[nodemon] restarting due to changes...
[nodemon] restarting due to changes...
[nodemon] starting `node --inspect=5858 index.js`
Debugger listening on ws://127.0.0.1:5858/d6fb72ad-ab3b-4c7c-9644-0ec3283c0000
For help, see: https://nodejs.org/en/docs/inspector
time="2020-02-11T01:29:26Z" level=debug msg="no auth required"
time="2020-02-11T01:29:26Z" level=info msg="stop process" program=run
time="2020-02-11T01:29:26Z" level=debug msg="succeed to find process:run"
time="2020-02-11T01:29:26Z" level=info msg="stop the program" program=run
time="2020-02-11T01:29:26Z" level=info msg="force to kill the program" program=run
time="2020-02-11T01:29:26Z" level=error msg="program stopped with error:signal: killed" program=run
time="2020-02-11T01:29:26Z" level=info msg="Stopped by user, don't start it again" program=run
time="2020-02-11T01:29:27Z" level=debug msg="no auth required"
time="2020-02-11T01:29:27Z" level=debug msg="succeed to find process:run"
time="2020-02-11T01:29:27Z" level=info msg="try to start program" program=run
time="2020-02-11T01:29:27Z" level=info msg="success to start program" program=run
+ set -eo pipefail
+ export ODO_UTILS_DIR=/opt/odo
+ ODO_UTILS_DIR=/opt/odo
+ export ODO_IMAGE_MAPPINGS_FILE=/opt/odo/language-scripts/image-mappings.json
+ ODO_IMAGE_MAPPINGS_FILE=/opt/odo/language-scripts/image-mappings.json
++ /opt/odo/bin/getlanguage
+ IMAGE_LANG=nodejs
+ '[' -f /tmp/.s2i/bin/run ']'
+ '[' '!' -z nodejs ']'
+ exec /opt/odo/language-scripts/nodejs/dev-run
+ DEV_MODE=true
+ '[' -z '' ']'
+ '[' true == true ']'
+ export NODE_ENV=development
+ NODE_ENV=development
+ '[' development == production ']'
+ export DEV_MODE=true
+ DEV_MODE=true
++ id -u
+ USER_ID=1008510000
++ id -g
+ GROUP_ID=0
+ '[' x1008510000 '!=' x0 -a x1008510000 '!=' x1001 ']'
+ NSS_WRAPPER_PASSWD=/opt/app-root/etc/passwd
+ NSS_WRAPPER_GROUP=/etc/group
+ sed -e 's/^default:/builder:/'
+ cat /etc/passwd
+ echo 'default:x:1008510000:0:Default Application User:/opt/app-root/src:/sbin/nologin'
+ export NSS_WRAPPER_PASSWD
+ export NSS_WRAPPER_GROUP
+ LD_PRELOAD=libnss_wrapper.so
+ export LD_PRELOAD
+ '[' development == production ']'
+ export DEV_MODE=true
+ DEV_MODE=true
+ set -x
+ export GIT_COMMITTER_NAME=unknown
+ GIT_COMMITTER_NAME=unknown
+ export and GIT_COMMITTER_EMAIL=unknown@localhost.com
+ GIT_COMMITTER_EMAIL=unknown@localhost.com
+ git --version
git version 1.8.3.1
+ '[' '' == --debug ']'
+ run_node
++ node --version
Using Node.js version: v10.16.3
Environment:
        DEV_MODE=true
        NODE_ENV=development
        DEBUG_PORT=5858
+ echo -e 'Using Node.js version: v10.16.3'
+ echo -e 'Environment: \n\tDEV_MODE=true\n\tNODE_ENV=development\n\tDEBUG_PORT=5858'
++ id
Running as user uid=1008510000(1008510000) gid=0(root) groups=0(root),1008510000
+ echo -e 'Running as user uid=1008510000(1008510000) gid=0(root) groups=0(root),1008510000'
+ '[' true == true ']'
+ echo 'Launching via nodemon...'
+ exec npx nodemon --inspect=5858
Launching via nodemon...
time="2020-02-11T01:29:28Z" level=debug msg="wait program exit" program=run
[nodemon] 2.0.2
[nodemon] to restart at any time, enter `rs`
[nodemon] watching dir(s): *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node --inspect=5858 index.js`
Debugger listening on ws://127.0.0.1:5858/af56d224-b6b1-4600-b71b-e4f567bdff20
For help, see: https://nodejs.org/en/docs/inspector
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module '/opt/app-root/src/index.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
[nodemon] app crashed - waiting for file changes before starting...
```

I didn't use this command the other day.

I did:

` oc get pods`

which returned 

```
NAME                                READY   STATUS      RESTARTS   AGE
nodejs-rhosapp7-ijey-app-1-deploy   0/1     Completed   0          18m
nodejs-rhosapp7-ijey-app-1-kz872    1/1     Running     0          18m
```

So completed and running statuses seemed like everything is fine.


The web ui for the pod says it is running:

![image](https://user-images.githubusercontent.com/5295985/74205545-d64c4b00-4c45-11ea-985c-cb1391704a99.png)

**Comment by dremond71** [comment]
```
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node --inspect=5858 index.js`
Debugger listening on ws://127.0.0.1:5858/af56d224-b6b1-4600-b71b-e4f567bdff20
For help, see: https://nodejs.org/en/docs/inspector
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module '/opt/app-root/src/index.js'
```

Perhaps this was caused by:

package.json
```
 "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "start": "node server.js"
  },
```
I start my app with : `npm start` 

but perhaps the "main" entry has to be `server.js`

I will try this.

**Comment by dremond71** [comment]
I deleted the project from web ui.
I started in a fresh directory
I made the change to main entry.

odo log 

now shows 

```
2020/02/11 02:08:20 [go-init] Pre-start command launched : /opt/odo/bin/s2i-setup
+ set -eo pipefail
+ PV_MNT_PT=/opt/app-root
+ ODO_UTILS_DIR=/opt/odo
+ '[' '!' -f /opt/app-root/conf/supervisor.conf ']'
+ cp -rp /opt/odo/conf /opt/app-root
+ mkdir -p /tmp
+ touch /tmp/.dummy
+ mkdir -p /tmp/src/
+ touch /tmp/src/.dummy
+ [[ /tmp != *\/\s\r\c* ]]
+ mkdir -p /tmp/src
+ touch /tmp/src/.dummy
+ mkdir -p /opt/app-root/src
2020/02/11 02:08:20 [go-init] Pre-start command exited
2020/02/11 02:08:20 [go-init] Main command launched : /opt/odo/bin/supervisord -c /opt/app-root/conf/supervisor.conf     
time="2020-02-11T02:08:20Z" level=info msg="create process:run"
time="2020-02-11T02:08:43Z" level=debug msg="no auth required"
time="2020-02-11T02:08:43Z" level=info msg="stop process" program=run
time="2020-02-11T02:08:43Z" level=debug msg="succeed to find process:run"
time="2020-02-11T02:08:43Z" level=info msg="stop the program" program=run
time="2020-02-11T02:08:43Z" level=info msg="force to kill the program" program=run
time="2020-02-11T02:08:43Z" level=debug msg="no auth required"
time="2020-02-11T02:08:43Z" level=debug msg="succeed to find process:run"
time="2020-02-11T02:08:43Z" level=info msg="try to start program" program=run
time="2020-02-11T02:08:43Z" level=info msg="success to start program" program=run
+ set -eo pipefail
+ export ODO_UTILS_DIR=/opt/odo
+ ODO_UTILS_DIR=/opt/odo
+ export ODO_IMAGE_MAPPINGS_FILE=/opt/odo/language-scripts/image-mappings.json
+ ODO_IMAGE_MAPPINGS_FILE=/opt/odo/language-scripts/image-mappings.json
++ /opt/odo/bin/getlanguage
+ IMAGE_LANG=nodejs
+ '[' -f /tmp/.s2i/bin/run ']'
+ '[' '!' -z nodejs ']'
+ exec /opt/odo/language-scripts/nodejs/dev-run
+ DEV_MODE=true
+ '[' -z '' ']'
+ '[' true == true ']'
+ export NODE_ENV=development
+ NODE_ENV=development
+ '[' development == production ']'
+ export DEV_MODE=true
+ DEV_MODE=true
++ id -u
+ USER_ID=1008790000
++ id -g
+ GROUP_ID=0
+ '[' x1008790000 '!=' x0 -a x1008790000 '!=' x1001 ']'
+ NSS_WRAPPER_PASSWD=/opt/app-root/etc/passwd
+ NSS_WRAPPER_GROUP=/etc/group
+ sed -e 's/^default:/builder:/'
+ cat /etc/passwd
+ echo 'default:x:1008790000:0:Default Application User:/opt/app-root/src:/sbin/nologin'
+ export NSS_WRAPPER_PASSWD
+ export NSS_WRAPPER_GROUP
+ LD_PRELOAD=libnss_wrapper.so
+ export LD_PRELOAD
+ '[' development == production ']'
+ export DEV_MODE=true
+ DEV_MODE=true
+ set -x
+ export GIT_COMMITTER_NAME=unknown
+ GIT_COMMITTER_NAME=unknown
+ export and GIT_COMMITTER_EMAIL=unknown@localhost.com
+ GIT_COMMITTER_EMAIL=unknown@localhost.com
+ git --version
git version 1.8.3.1
+ '[' '' == --debug ']'
+ run_node
++ node --version
Using Node.js version: v10.16.3
+ echo -e 'Using Node.js version: v10.16.3'
+ echo -e 'Environment: \n\tDEV_MODE=true\n\tNODE_ENV=development\n\tDEBUG_PORT=5858'
Environment:
        DEV_MODE=true
        NODE_ENV=development
        DEBUG_PORT=5858
++ id
+ echo -e 'Running as user uid=1008790000(1008790000) gid=0(root) groups=0(root),1008790000'
+ '[' true == true ']'
+ echo 'Launching via nodemon...'
+ exec npx nodemon --inspect=5858
Running as user uid=1008790000(1008790000) gid=0(root) groups=0(root),1008790000
Launching via nodemon...
time="2020-02-11T02:08:44Z" level=debug msg="wait program exit" program=run
[nodemon] 2.0.2
[nodemon] to restart at any time, enter `rs`
[nodemon] watching dir(s): *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node --inspect=5858 server.js`
Debugger listening on ws://127.0.0.1:5858/dc2e88a0-3732-4823-b70a-b2fe6ab18aa1
For help, see: https://nodejs.org/en/docs/inspector
server running on port 8080
[nodemon] restarting due to changes...
[nodemon] restarting due to changes...
[nodemon] restarting due to changes...
[nodemon] starting `node --inspect=5858 server.js`
Debugger listening on ws://127.0.0.1:5858/a2017fd5-a2ab-4383-b910-3de59e19606f
For help, see: https://nodejs.org/en/docs/inspector
[nodemon] restarting due to changes...
[nodemon] restarting due to changes...
[nodemon] restarting due to changes...
[nodemon] starting `node --inspect=5858 server.js`
[nodemon] restarting due to changes...
Debugger listening on ws://127.0.0.1:5858/84ecfc38-3d3f-412f-97d6-4dbd0e3116c8
For help, see: https://nodejs.org/en/docs/inspector
[nodemon] restarting due to changes...
[nodemon] restarting due to changes...
[nodemon] starting `node --inspect=5858 server.js`
Debugger listening on ws://127.0.0.1:5858/709ffd2f-f5a2-45be-8bf9-0391e5def59b
For help, see: https://nodejs.org/en/docs/inspector
time="2020-02-11T02:09:43Z" level=debug msg="no auth required"
time="2020-02-11T02:09:43Z" level=info msg="stop process" program=run
time="2020-02-11T02:09:43Z" level=debug msg="succeed to find process:run"
time="2020-02-11T02:09:43Z" level=info msg="stop the program" program=run
time="2020-02-11T02:09:43Z" level=info msg="force to kill the program" program=run
time="2020-02-11T02:09:43Z" level=error msg="program stopped with error:signal: killed" program=run
time="2020-02-11T02:09:43Z" level=info msg="Stopped by user, don't start it again" program=run
time="2020-02-11T02:09:44Z" level=debug msg="no auth required"
time="2020-02-11T02:09:44Z" level=debug msg="succeed to find process:run"
time="2020-02-11T02:09:44Z" level=info msg="try to start program" program=run
time="2020-02-11T02:09:44Z" level=info msg="success to start program" program=run
+ set -eo pipefail
+ export ODO_UTILS_DIR=/opt/odo
+ ODO_UTILS_DIR=/opt/odo
+ export ODO_IMAGE_MAPPINGS_FILE=/opt/odo/language-scripts/image-mappings.json
+ ODO_IMAGE_MAPPINGS_FILE=/opt/odo/language-scripts/image-mappings.json
++ /opt/odo/bin/getlanguage
+ IMAGE_LANG=nodejs
+ '[' -f /tmp/.s2i/bin/run ']'
+ '[' '!' -z nodejs ']'
+ exec /opt/odo/language-scripts/nodejs/dev-run
+ DEV_MODE=true
+ '[' -z '' ']'
+ '[' true == true ']'
+ export NODE_ENV=development
+ NODE_ENV=development
+ '[' development == production ']'
+ export DEV_MODE=true
+ DEV_MODE=true
++ id -u
+ USER_ID=1008790000
++ id -g
+ GROUP_ID=0
+ '[' x1008790000 '!=' x0 -a x1008790000 '!=' x1001 ']'
+ NSS_WRAPPER_PASSWD=/opt/app-root/etc/passwd
+ NSS_WRAPPER_GROUP=/etc/group
+ cat /etc/passwd
+ sed -e 's/^default:/builder:/'
+ echo 'default:x:1008790000:0:Default Application User:/opt/app-root/src:/sbin/nologin'
+ export NSS_WRAPPER_PASSWD
+ export NSS_WRAPPER_GROUP
+ LD_PRELOAD=libnss_wrapper.so
+ export LD_PRELOAD
+ '[' development == production ']'
+ export DEV_MODE=true
+ DEV_MODE=true
+ set -x
+ export GIT_COMMITTER_NAME=unknown
+ GIT_COMMITTER_NAME=unknown
+ export and GIT_COMMITTER_EMAIL=unknown@localhost.com
+ GIT_COMMITTER_EMAIL=unknown@localhost.com
+ git --version
git version 1.8.3.1
+ '[' '' == --debug ']'
+ run_node
++ node --version
Using Node.js version: v10.16.3
+ echo -e 'Using Node.js version: v10.16.3'
Environment:
        DEV_MODE=true
        NODE_ENV=development
        DEBUG_PORT=5858
+ echo -e 'Environment: \n\tDEV_MODE=true\n\tNODE_ENV=development\n\tDEBUG_PORT=5858'
++ id
Running as user uid=1008790000(1008790000) gid=0(root) groups=0(root),1008790000
+ echo -e 'Running as user uid=1008790000(1008790000) gid=0(root) groups=0(root),1008790000'
+ '[' true == true ']'
+ echo 'Launching via nodemon...'
+ exec npx nodemon --inspect=5858
Launching via nodemon...
time="2020-02-11T02:09:45Z" level=debug msg="wait program exit" program=run
[nodemon] 2.0.2
[nodemon] to restart at any time, enter `rs`
[nodemon] watching dir(s): *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node --inspect=5858 server.js`
Debugger listening on ws://127.0.0.1:5858/ffaaa102-a9fe-4923-8f9b-5a2b30e2ddc9
For help, see: https://nodejs.org/en/docs/inspector
server running on port 8080
```

No errors.

`oc get route`

returns 

```
 oc get route
NAME                            HOST/PORT                                                                            PATH   SERVICES       
            PORT   TERMINATION   WILDCARD
nodejs-rhosapp3-vadp-8080-app   nodejs-rhosapp3-vadp-8080-app-rhosapp3.apps.us-east-1.starter.openshift-online.com          nodejs-rhosapp3-vadp-app   8080                 None
```

**Comment by dremond71** [comment]
`oc get svc`

```
NAME                       TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
nodejs-rhosapp3-vadp-app   ClusterIP   172.30.166.46   <none>        8080/TCP   20m
```

**Comment by dremond71** [comment]
so... exactly how do I access my application?

With this url?

http://nodejs-rhosapp3-vadp-8080-app-rhosapp3.apps.us-east-1.starter.openshift-online.com:8080  

?



In postman

POST http://nodejs-rhosapp3-vadp-8080-app-rhosapp3.apps.us-east-1.starter.openshift-online.com:8080/api/v1/grammarcheck 

just hangs

Ping    http://nodejs-rhosapp3-vadp-8080-app-rhosapp3.apps.us-east-1.starter.openshift-online.com 
times out.

**Comment by dremond71** [comment]
oh

if I use 

http://nodejs-rhosapp3-vadp-8080-app-rhosapp3.apps.us-east-1.starter.openshift-online.com/api/v1/myendpoint 

without specifying :8080 , then it works.

Ok. that is unexpected.

I figured for sure I would need to use a port.

I can access my endpoint now.

Is this documented? That it is not necessary to use the port portion of the url?

**Comment by kadel** [comment]
> oh
> 
> if I use
> 
> http://nodejs-rhosapp3-vadp-8080-app-rhosapp3.apps.us-east-1.starter.openshift-online.com/api/v1/myendpoint
> 
> without specifying :8080 , then it works.
> 
> Ok. that is unexpected.
> 
> I figured for sure I would need to use a port.
> 
> I can access my endpoint now.
> 
> Is this documented? That it is not necessary to use the port portion of the url?

Odo shows you the full and correct url in the `odo push` output.
The same for `odo url list` the URL displayed there is the one that you should use.

URL is represented as a Router OpenShift Resource, you can't really use a different port than default ones( 80 for http and 443 for https.)

**Comment by kadel** [comment]
closing this one
@dremond71 if you find any issues with odo, feel free to reopen this or open a new issue

/close
